humps.js

```diff
@@ -40,26 +40,29 @@
   // String conversion methods
 
   var separateWords = function(string, separator) {
-    if (separator === undefined) {
+    if (typeof separator === 'undefined') {
       separator = '_';
     }
     return string.replace(/([a-z])([A-Z0-9])/g, '$1'+ separator +'$2');
   };
 
   var camelize = function(string) {
+    // efficiently checks if a key is a number
+    var number = string - 0;
+    if (number === number) {
```

Comment:
Well, there's isNaN native function which would work simillary but its 100% slower: see https://jsperf.com/isnan-vs-quick-nan-check . We might create _isNaN function and insert the code in there, what do you say?